src/main/g8/src/main/scala/$package$/DemoParallelism.scala

```diff
@@ -1,48 +1,75 @@
 package $package$
 
+import cats.effect.Resource
 import com.typesafe.scalalogging.LazyLogging
 import monix.eval.Task
 import monix.execution.Scheduler
+import monix.reactive.Observable
 
-import scala.concurrent.Await
 import scala.concurrent.duration._
 import scala.util.Random
 
 object DemoParallelism extends LazyLogging {
 
   /**
-    * Demonstrates effective parallelism using Monix and Monte-Carlo pi calculation.
-    *
-    * @param iterations  Iterations to calculate pi for
-    * @param parallelism Concurrent processes
-    */
-  def apply(iterations: Long, parallelism: Int): Unit = {
-    logger.info(f"Will approximate Pi with \$iterations%,d iterations of random bubble-filling and \$parallelism%,d parallel processes...")
+   * Demonstrates effective parallelism using Monix and Monte-Carlo pi calculation.
+   *
+   * @param iterations  Iterations to calculate pi for
+   * @param parallelism Concurrent processes
+   */
+  def apply(iterations: Long, parallelism: Int): Task[Double] = for {
+    // log that the work is starting
+    _ <- Task {
+      logger.info(f"Will approximate Pi with \$iterations%,d iterations of random bubble-filling and up to \$parallelism%,d parallel processes...")
+    }
 
-    // Scheduler for parallelism:
-    implicit val s: Scheduler = Scheduler.forkJoin(parallelism, maxThreads = 1000)
+    // create scheduler resource based on desired parallelism (it will be closed once we have finished our calculations)
+    schedulerResource = Resource.make(acquire = Task {
+      Scheduler.forkJoin(parallelism = parallelism, maxThreads = 1000)
+    })(release = s => Task { s.shutdown() })
```

Comment:
This thing would still exit just fine without shutting down this scheduler, but I like the fact that this function is now guaranteed to clean up after itself. That's nice if someone ever decides to copy pasta it into another project.

I'm always happy to advertise how awesome `cats.effect.Resource` is, but I concede that it is a bit off-topic for this project. For that reason, I'm on the fence about whether it's worth keeping around.

Reply:
Keep it!

I learned something reading this, and the point of this template is to establish good and sane practices, _(and then let people willingly delete those good things)_.